models/Goal.model.js

```diff
@@ -0,0 +1,28 @@
+
+const { Schema, model } = require("mongoose");
+
+const userSchema = new Schema(
+    {
+        name:{
+            type: String,
+            required: true
+        },
+        key : {
+            type : String,
+            required: true,
+        },
```

Comment:
what is the meaning of key for goal?

Reply:
This is a field entered by the client to then access the specific goal once there are multiple goals.
for example, a client can have several goals with a goal where the key equals "login" stating that the end user had reached the login page, and another goal that has the key "purchase" for confirming a purchase

Reply:
So you are creating a goal groups of a sort? Or is it meant to be uniqe?

Reply:
You are right it is supposed to be unique value, made the changes.